File: 2044-count-number-of-maximum-bitwise-or-subsets/2044-count-number-of-maximum-bitwise-or-subsets.py
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        OR = 0
        for x in nums:
            OR = OR | x
        count = [0]
        def solve(i: int, ans: int):
            if i >= len(nums):
                if ans == OR:
                    count[0] += 1
                return
            solve(i + 1, ans | nums[i])
            solve(i + 1, ans)        
        solve(0, 0)
        return count[0] 

File: 2044-count-number-of-maximum-bitwise-or-subsets/2044-count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int OR;
    int count = 0;
    void solve(int i, vector<int> &nums, int ans) {
        if (nums.size() <= i) {
            if (ans == OR) {
                count++;
            }
            return;
        }
        solve(i+1, nums, ans | nums[i]);
        solve(i+1, nums, ans);
    }
    int countMaxOrSubsets(vector<int>& nums) {
        OR = 0;
        for (int x : nums) OR = OR | x;
        solve(0, nums, 0);
        return count;
    }
};